### PR TITLE
Feature/global segments

### PR DESCRIPTION
**Pull request by sighphyre**
This adds support for reusable global segments, reference implementation in node can be viewed here: https://github.com/Unleash/unleash-client-node/pull/329

This also includes the header for turning on segments

### CODE REVIEW

File: UnleashClient/variants/Variants.py [hunk 1]
```diff
@@ -79,7 +79,7 @@ def get_variant(self, context: dict) -> dict:
             if override_variant:
                 return self._format_variation(override_variant)
 
-            total_weight = sum([x['weight'] for x in self.variants])
+            total_weight = sum(x['weight'] for x in self.variants)
```
Comment (sighphyre) [inline UnleashClient/variants/Variants.py:82]: Swaps a list comprehension for a generator because Pylint doesn't like this

File: pyproject.toml [hunk 2]
```diff
@@ -15,6 +15,9 @@ addopts= """
 """
 log_file_level="INFO"
 
+[tool.pylint]
+load-plugins="pylint.extensions.no_self_use"
```
Comment (sighphyre) [inline pyproject.toml:19]: Pylint got updated, the no_self_use option is now opt in so this turns that back on

**Review by chriswk — APPROVED** [review]
The logic seems sound, but have to admit my Python is quite rusty, the implementation seems close to how we do it in Node and Java

**Comment by chriswk** [comment]
[Segmentation SDK support: Python](https://trello.com/c/vDW3UZxF/1285-segmentation-sdk-support-python)

**Comment by ivanklee86** [comment]
Could I get a link to a document outlining Segmentation (or maybe the Trello ticket)?  Just wanna 👀 to make sure I understand what's supposed to be happening. 😄 

(Sorry, busy this week 'cause going on PTO next week.  Should offer more feedback

**Pull request merged**